plugin-oar2/src/lib.rs

```diff
@@ -0,0 +1,286 @@
+            metrics: None,
+            watcher: None,
+        }))
+    }
+
+    fn start(&mut self, alumet: &mut AlumetStart) -> Result<(), anyhow::Error> {
+        let cpu_metric = alumet.create_metric::<u64>(
+            "cpu_time",
+            PrefixedUnit::nano(Unit::Second),
+            "Total CPU time consumed by the cgroup (in nanoseconds).",
+        )?;
+        let memory_metric = alumet.create_metric::<u64>(
+            "memory_usage",
+            Unit::Unity,
+            "Total memory usage by the cgroup (in bytes).",
+        )?;
+
+        self.metrics = Some(Metrics {
+            cpu_metric,
+            memory_metric,
+        });
+
+        let cgroup_cpu_path = self.config.path.join("cpuacct/oar");
+        let cgroup_memory_path = self.config.path.join("memory/oar");
+
+        // Scanning to check if there are jobs already running
+        for entry in fs::read_dir(&cgroup_cpu_path)
+            .with_context(|| format!("Invalid oar cpuacct cgroup path, {cgroup_cpu_path:?}"))?
+        {
+            let entry = entry?;
+
+            let job_name = entry.file_name();
+            let job_name = job_name.into_string().ok().with_context(|| format!("Invalid oar username and job id."))?;
+
+            if entry.file_type()?.is_dir() && job_name.chars().any(|c| c.is_numeric()) {
+                let job_separated = job_name.split_once("_");
+                let job_id = job_separated.context("Invalid oar cgroup.")?.1.parse()?;
+
+                let cpu_job_path = cgroup_cpu_path.join(&job_name);
+                let memory_job_path = cgroup_memory_path.join(&job_name);
+
+                let cpu_file_path = cpu_job_path.join("cpuacct.usage");
+                let memory_file_path = memory_job_path.join("memory.usage_in_bytes");
+
+                let cgroup_cpu_file = File::open(&cpu_file_path)
+                    .with_context(|| format!("Failed to open CPU usage file at {}", cpu_file_path.display()))?;
+                let cgroup_memory_file = File::open(&memory_file_path)
+                    .with_context(|| format!("Failed to open memory usage file at {}", memory_file_path.display()))?;
+
+                let initial_source = Box::new(OarJobSource {
+                    cpu_metric,
+                    memory_metric,
+                    cgroup_cpu_file,
+                    cgroup_memory_file,
+                    cpu_file_path,
+                    memory_file_path,
+                    job_id,
+                });
+
+                alumet.add_source(initial_source, TriggerSpec::at_interval(self.config.poll_interval));
+            }
+        }
+        Ok(())
+    }
+
+    fn post_pipeline_start(&mut self, pipeline: &mut alumet::pipeline::runtime::RunningPipeline) -> anyhow::Result<()> {
+        let control_handle = pipeline.control_handle();
+        let config_path = self.config.path.clone();
+        let plugin_name = self.name().to_owned();
+
+        let metrics = self.metrics.clone().unwrap();
```

Comment:
```suggestion
        let metrics = self.metrics.take().expect("metrics is set in start()");
```

self.metrics cannot be None here